tests/kafkatest/services/security/minikdc.py

```diff
@@ -39,14 +40,48 @@ class MiniKdc(KafkaPathResolverMixin, Service):
     KEYTAB_FILE = "/mnt/minikdc/keytab"
     KRB5CONF_FILE = "/mnt/minikdc/krb5.conf"
     LOG_FILE = "/mnt/minikdc/minikdc.log"
-    LOCAL_KEYTAB_FILE = "/tmp/" + str(uuid.uuid4().get_hex()) + "_keytab"
-    LOCAL_KRB5CONF_FILE = "/tmp/" + str(uuid.uuid4().get_hex()) + "_krb5.conf"
+
+    LOCAL_KEYTAB_FILE = None
+    LOCAL_KRB5CONF_FILE = None
+
+    @staticmethod
+    def _set_local_keytab_file(local_scratch_dir):
```

Comment:
Yeah, initially I was hoping to do something along those lines but threading the `minikdc` instance throughout seemed painful. 

On second look, the fallout does seem manageable - the only services which instantiate minikdc are `KafkaService` and `MirrorMaker`, and it looks like all services that need it have access to a `self.kafka` or a `minikdc` instance.

I'll give the change a shot and see if the scope is what I think it is.